Add pdf-renderer

IldarMinaev opened the pull request:
- Link the PR to an issue for new features
- Rebase your PR if it gets out of sync with master
  Requests /getpdf/\* will return PDF rendered files

IldarMinaev committed:
Add pdf-renderer

Requests /getpdf/* will return PDF rendered files

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=6028) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=6028) before we can accept your contribution.<br/>

torkelo commented:
a bit too much duplication here from the png render code :)

IldarMinaev committed:
Revert "Add pdf-renderer"

This reverts commit 3097b047f1778162024206b8bd133af17a7cce47.

IldarMinaev committed:
Merge remote-tracking branch 'refs/remotes/grafana/master'

IldarMinaev committed:
Add file type to renderer

Add file type for render:
/render/pdf/dashboard... will create pdf of dashboard.
/rendre/jpg/dashboard... - create jpeg
/rendre/png/dashboard... - create png
/render/dashboard... - create png

IldarMinaev committed:
Formating

IldarMinaev commented:
Hi @torkelo, I have made some changes in new commit. Now there is one function RenderToFile, which have second argument type of generating snapshot. It support formats which can be created by phantomJS: pdf, png, bmp, jpg, ppm.
To select format you need put your format after 'render' in URL path:
grafanahost:port/render/pdf/dashboard/db/new-dashboard

Selection of output format supported only for dashboards URLs.

I am new for github and go, and have question about tests. How to pass this one?
$ test -z "$(gofmt -s -l . | grep -v Godeps/_workspace/src/ | tee /dev/stderr)" [#5169](https://circleci.com/gh/grafana/grafana/5169#tests/containers/0)

IldarMinaev committed:
Change render logic to wait all panels rendered.

IldarMinaev committed:
Fix render.js to properly check panels loaded

torkelo commented:
closing due to lack of interest / feature request. 

Created feature request where users can vote for this feature and describe the need for it:
https://github.com/grafana/grafana/issues/6843
